A3A/addons/scrt/Encounter/fn_encounter_fleeingCivConvoy.sqf

```diff
@@ -0,0 +1,162 @@
+#include "..\defines.inc"
+FIX_LINE_NUMBERS()
+
+Info( "Vehicle Move Event Init.");
+
+private _vehicles = [];
+private _groups = [];
+
+private _player = selectRandom (call SCRT_fnc_misc_getRebelPlayers);
+
+if (isNil "_player") exitWith {
+    Error("No players found, aborting.");
+    isEventInProgress = false;
+    publicVariableServer "isEventInProgress";
+};
+
+private _originPosition = position _player;
+
+Info_2("%1 will be used as center of the event at %2 position.", name _player, str _originPosition);
+
+private _city = if (_originPosition isEqualType "") then {_originPosition} else {[citiesX, _originPosition] call BIS_fnc_nearestPosition};
+
+if (_city isEqualTo []) exitWith {
+    Info("No cities in proximity, aborting civ convoy Event.");
+    isEventInProgress = false;
+    publicVariableServer "isEventInProgress";
+};
+
+private _anothercity = selectRandom citiesX;
+while {true} do {
+	if (_anothercity != _city) exitwith{};
+	private _anothercity = selectRandom citiesX;
+};
```

Comment:
```suggestion
private _anotherCity = selectRandom (citiesX select {_x != _city})
```

Can't this achieve the same thing?
Make sure to test it as i'm not entirely sure, but these rogue while loops are terrible to use